src/Factory/HalControllerPluginFactory.php

```diff
@@ -7,17 +7,36 @@
 namespace ZF\Hal\Factory;
 
 use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\AbstractPluginManager;
+use Zend\ServiceManager\FactoryInterface;
+use Zend\ServiceManager\ServiceLocatorInterface;
 use ZF\Hal\Plugin\Hal;
 
-class HalControllerPluginFactory
+class HalControllerPluginFactory implements FactoryInterface
 {
     /**
-     * @param ContainerInterface $container
+     * @param  ContainerInterface $container
+     * @param  string             $requestedName
+     * @param  null|array         $options
      * @return Hal
      */
-    public function __invoke(ContainerInterface $container)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         $helpers  = $container->get('ViewHelperManager');
         return $helpers->get('Hal');
     }
+
+    /**
+     * Create service
+     *
+     * @param ServiceLocatorInterface $serviceLocator
+     * @return Hal
+     */
+    public function createService(ServiceLocatorInterface $serviceLocator)
+    {
+        if ($serviceLocator instanceof AbstractPluginManager) {
+            $serviceLocator = $serviceLocator->getServiceLocator();
```

Comment:
This should likely be `$serviceLocator = $serviceLocator->getServiceLocator() ?: $serviceLocator;`. I'll make that change on merge.